src/lib.rs

```diff
@@ -0,0 +1,52 @@
+//! This crate provides a single macro to help skipping a error in a loop,
+//! possibly logging it.
+//!
+//! For example, imagine you have some code like this.
+//! ```edition2018
+//! for string_number in &["1", "2", "three", "4"] {
+//!   let number: u32 = match string_number.parse() {
+//!     Ok(n) => n,
+//!     Err(e) => continue,
+//!   };
+//! }
+//! ```
+//!
+//! Then you can use the macro `skip_fail!` to write like this.
+//! ```edition2018
+//! # #[macro_use]
+//! # extern crate skip_fail;
+//! # fn main() {
+//! for string_number in &["1", "2", "three", "4"] {
+//!   let number: u32 = skip_fail!(string_number.parse());
+//! }
+//! # }
+//! ```
+//!
+//! If you want the error to be logged, you can use the feature `log`. The
+//! logging will be done in WARN level with the standard logging interface
+//! provided by [`log`](https://crates.io/crates/log).
+#[macro_export]
+#[cfg(not(feature = "log"))]
```

Comment:
interesting but I think it would better if we can use the 2 macros at the sam time

Reply:
I like this idea. It didn't even crossed my mind. I see 2 ways to achieve this functionality:
- adding a parameter to the macro (a boolean `with_log`)
- creating 2 macros named differently (we could name it `skip_fail_with_warn`)

I'm much more in favor of the second solution (we could even think of a third one later with `skip_fail_with_info`).

What do you think?

Reply:
I prefer `skip_fail_with_warn` but a bit different
WHy not something like `skip_fail_and_log` with a second optional parameter `log_level`?

Reply:
I'm fine with that, I'll do that.

Reply:
I've added the macro `skip_fail_and_log` under the feature `log`. This macro now takes 2 parameters, the `Result` and a `log::Level` (which is an enum). I also added more documentation, for each macro.